### PR TITLE
Add link to swap branches on Compare

### LINKED ISSUE #1217: Button to switch base and head in "Compare" page
I suggest to implement a function to switch the base and the header fork positions with each other into this currently unused icon:

http://funkyimg.com/view/2DKTW

On this way, it is easy to do git pull, as an example. :slightly_smiling_face:

### PR DESCRIPTION
**Pull request by busches**
Closes #1217.

Tested on the following URLs:
https://github.com/sindresorhus/refined-github/compare/master...no-releases
https://github.com/sindresorhus/refined-github/compare/no-releases...master
https://github.com/sindresorhus/refined-github/compare/no-releases
https://github.com/sindresorhus/refined-github/compare/master...busches:fix-console-errors
https://github.com/busches/refined-github/compare/fix-console-errors...sindresorhus:master

### CODE REVIEW

File: source/features/add-switch-branches-on-compare.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+	// Compares against the "base" branch only have one in the URL
+	const currentBase = compareURL.length > 1 ? compareURL[0] : select('.branch span').textContent;
+	const currentCompare = compareURL[compareURL.length - 1];
+
+	const link = <a href={`/${pageDetect.getRepoURL()}/compare/${currentCompare}...${currentBase}`}></a>;
+
+	// Wrap the icon in our new link
+	compareIcon.parentNode.insertBefore(link, compareIcon);
+	link.appendChild(compareIcon);
```
Comment (bfred-it) [inline source/features/add-switch-branches-on-compare.js]: There's a dedicated `wrap` function in `utils.js`

File: readme.md [hunk 2]
```diff
@@ -159,6 +159,7 @@ GitHub Enterprise is also supported. More info in the options.
 - [Dropdown menus are automatically closed when they’re no longer visible.](https://user-images.githubusercontent.com/1402241/37022353-531c676e-2155-11e8-96cc-80d934bb22e0.gif)
 - [Permalinked comments are highlighted.](https://user-images.githubusercontent.com/1402241/37349492-226bd37a-2709-11e8-8087-d9686b330240.png)
 - [Inactive deployments in PR timelines are hidden.](https://github.com/sindresorhus/refined-github/issues/1144)
+- Link to quickly swap branches when doing a branch compare.
```
Comment (bfred-it) [inline readme.md]: This should probably be in the _actions_ sections with wording like "Swap branches in the branch compare view"

File: source/features/add-switch-branches-on-compare.js (hunk 1 above)
Comment (busches) [reply source/features/add-switch-branches-on-compare.js]: Not sure how I missed that, figured we'd have one.

File: readme.md (hunk 2 above)
Comment (busches) [reply readme.md]: I couldn't think of good wording, thanks.

**Comment by bfred-it** [comment]
What do you think about these changes? As suggested by @ShalokShalom 

screen shot 2018-06-05 at 10 19 14

**Comment by busches** [comment]
I liked it on the icon, since it looks like a swap icon. Is there a world where I convince you to have both? ;)

**Comment by ShalokShalom** [comment]
Both is nice, imho. And yes: The tooltip is super awesome

**Comment by bfred-it** [comment]
>  it looks like a swap icon

We probably shouldn't conflate the two meanings.

> have both?

I think there should only be a way to do it:

- either between the two (as many travel sites show it) or
- as an extra, clear button on the side (with a [different icon](https://user-images.githubusercontent.com/1402241/42055570-cffda8ee-7b17-11e8-95fa-9d648d664d51.png) or just text)

After the readme is updated, we can merge this

**Comment by ShalokShalom** [comment]
I am pro with the "travel-sites" argument: Humans already know this, so pick us up where we are?

File: source/features/add-swap-branches-on-compare.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+export default async () => {
+	const references = getRepoPath().replace('compare/', '').split('...').reverse();
+
+	// Compares against the "base" branch if the URL only has one reference
+	if (references.length === 1) {
+		references.unshift(select('.branch span').textContent);
+	}
+
+	const icon = select('.octicon-arrow-left');
+	icon.parentNode.attributes['aria-label'].value += '. Click to swap.';
```
Comment (sindresorhus) [inline source/features/add-swap-branches-on-compare.js]: Would look better if `Click to swap.` was on a new line.

`'. Click to swap.'` => `'.\nClick to swap.'`

**Comment by busches** [comment]
Uploading screenshot 
image

**Pull request merged**